iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi_process.hpp

```diff
@@ -206,7 +213,39 @@ class ProcessManager : public ProcessManagerInterface
                     const uint64_t sessionId,
                     const version::VersionInfo& versionInfo) noexcept;
 
+    /// @brief Removes the process from the managed client process list, identified by its id.
+    /// @param [in] name The process name which should be removed.
+    /// @return Returns true if the process was found and removed from the internal list.
     bool removeProcess(const ProcessName_t& name) noexcept;
+
+    /// @brief Removes the given process from the managed client process list without taking the list's lock!
+    /// @param [in] lockGuard This method has to be called within a lock guard context. By providing this lock guard it
+    ///                       ensures it can't be called without a lock guard in place.
+    /// @param [in] processIter The process which should be removed.
+    /// @return Returns true if the process was found and removed from the internal list.
+    bool removeProcess(const std::lock_guard<std::mutex>& lockGuard, ProcessList_t::iterator& processIter) noexcept;
+
+    enum class ShutdownPolicy
+    {
+        SIG_TERM,
+        SIG_KILL
+    };
+
+    enum class ShutdownLog
+    {
+        NONE,
+        FULL
+    };
+
+    /// @brief Kills the given process in m_processList with the given signal.
```

Comment:
A bit misleading... it's not killing unless `shutdownPolicy == ShutdownPolicy.SIG_KILL`.

Reply:
Renamed all kills.